### PR TITLE
[DOC] Add Nested function SQL documentation  - autoclosed

### PR DESCRIPTION
**Pull request by vagimeli**
### Description
Copy dev authored content to new branch and perform doc review; creates nested function doc

### Issues Resolved
See PR https://github.com/opensearch-project/documentation-website/pull/4308 

### Checklist
- [X] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by vagimeli** [comment]
Per communication with @anirudha, this content should go with how to work with JSON data. See the following notes:

- https://github.com/opensearch-project/sql/blob/main/docs/user/index.rst all SQL / PPL detailed docs and references are here. They are auto generated per build
- nested docs are here : https://github.com/opensearch-project/sql/blob/main/docs/user/beyond/partiql.rst
- PPL is here : https://github.com/opensearch-project/sql/blob/main/docs/user/ppl/index.rst

**Comment by penghuo** [comment]
looks good to me.

File: _search-plugins/sql/sql/nested-function.md [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+The syntax is shown in the following example:
+
+```sql
+nested(field_expression | field_expression, path_expression)
+```
+{% include copy.html %}
+
+#### Example query 
+
+The following are examples of a query using nested queries in the `GROUP BY` and `HAVING` clauses:
```
Comment (natebower) [inline _search-plugins/sql/sql/nested-function.md:207]: "The following example query uses" (singular)?

File: _search-plugins/sql/sql/nested-function.md [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+This option, which is supported by the V2 engine, places the `nested` function on the left of the predicate expression and on the right of the `literal_expression`. The `path_expression` is optional and automatically determined by the SQL plugin if not provided, as shown in the following example:
+
+```sql
+nested(field_expression | field_expression, path_expression) Operator Literal_expression
+```
+{% include copy.html %}
+
+#### Example query
+
+The following are examples of a query using nested queries in the `WHERE` clause, a sample dataset, and the query results:
```
Comment (natebower) [inline _search-plugins/sql/sql/nested-function.md:118]:
```suggestion
The following are examples of a query using nested queries in the `WHERE` clause:
```

File: _search-plugins/sql/sql/nested-function.md [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+The `ORDER BY` clause with the `nested` function defaults to ascending order (`ASC`). Specifying `DESC` (desending) after the function allows overriding this behavior. The syntax is shown in the following example: 
+
+```sql
+nested(field_expression | field_expression, path_expression)
+```
+{% include copy.html %}
+
+#### Example query 
+
+The following are examples of a query using nested queries in the `ORDER BY` clause, a sample dataset, and the query results:
```
Comment (natebower) [inline _search-plugins/sql/sql/nested-function.md:163]:
```suggestion
The following are examples of a query using nested queries in the `ORDER BY` clause:
```

**Review by natebower — COMMENTED** [review]
@vagimeli Please see my comments and changes and let me know if you have any questions. Thanks!

File: _search-plugins/sql/sql/nested-function.md [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+
+The `nested` function is used to created a query within a query. See the following sections for how and why to use the function in OpenSearch.   
+
+## Using the function in a SELECT clause
+
+To extract individual elements from nested object type collections within the `SELECT` clause, use the `nested` function. This process flattens nested structures, generating a Cartesian product when querying against nested collections. 
+
+### Syntax
+
+The `field_expression` parameter is required. The `path_expression` parameter is optional. Dot notation is used to show the nesting level for both the `field_expression` and `path_expression` parameters. For example, `nestedObj.innerFieldName` denotes a field nested one level. If the user does not provide the `path_expression` parameter, the path value is generated dynamically. For example, the field `user.office.cubicle` would dynamically generate the path `user.office`. The syntax is shown in the following example:
```
[the remark this answers is not in the document]
Comment (vagimeli) [reply _search-plugins/sql/sql/nested-function.md:22]: Revised to read "nested one level deep"

File: _search-plugins/sql/sql/nested-function.md [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+This option specifies the Boolean condition inside the nested function using the `condition_expression` parameter, as shown in the following example:
+
+```sql
+nested(path_expression, condition_expression)
+```
+{% include copy.html %}
+
+### Nested function with predicate expression and literal expression
+
+This option, which is supported by the V2 engine, places the `nested` function on the left of the predicate expression and on the right of the `literal_expression`. The `path_expression` is optional and automatically determined by the SQL plugin if not provided, as shown in the following example:
```
[the remark this answers is not in the document]
Comment (vagimeli) [reply _search-plugins/sql/sql/nested-function.md:109]: It's regular typeface. It's a common noun in this case.

[the remark this answers is not in the document]
Comment (vagimeli) [reply _search-plugins/sql/sql/nested-function.md:108]: @penghuo Should _predicate expression_ be `predicate_expression`? Should `literal_expression` be `Literal_expression` as written in the example below?

**Comment by vagimeli** [comment]
> looks good to me.

Hi @penghuo: When I tested the query examples, I got the following error. Will you please review and let me know if this error is my user error or if we need to revise the query examples? Thank you, @vagimeli 
 
Screenshot 2024-01-11 at 10 38 19 AM

**Pull request closed without merging**